onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -3133,14 +3253,25 @@ Status TensorrtExecutionProvider::CreateNodeComputeInfoFromGraph(const GraphView
       cache_path = GetCachePath(trt_state->engine_cache_path, trt_state->trt_node_name_with_precision);
     }
     const std::string cache_path_prefix = cache_path + "_sm" + compute_capability_;
-    const std::string engine_cache_path = cache_path_prefix + ".engine";
+    std::string engine_cache_path = cache_path_prefix + ".engine";
     const std::string encrypted_engine_cache_path = engine_cache_path + ".encrypted";
     const std::string profile_cache_path = cache_path_prefix + ".profile";
     std::string timing_cache_path = "";
     if (timing_cache_enable_) {
       timing_cache_path = GetTimingCachePath(global_cache_path_, compute_capability_);
     }
 
+    // TRT EP uses the engine cache with ".stripped.engine" appended to the end if following three conditions are all true:
```

Comment:
nit: Can we also make this a lambda given lines 2915-2925?

Reply:
Plan to have another PR to do this as well as the "fill cache paths" code above in the same lambda function.